### ISSUE TITLE
Unable to use Key Vault from Common Infra in multiple environments

### ISSUE THREAD

**Issue by sarath-p** [microsoft/bedrock#854]
**Describe the bug:**

Unable to use a single key vault from common infra in multiple environments.

```
module.keyvault_flexvolume_role.azurerm_role_assignment.flexvol[0]: Creating...
module.aks-gitops.module.aks.local_file.cluster_credentials[0]: Creating...
module.aks-gitops.module.aks.local_file.cluster_credentials[0]: Creation complete after 0s [id=b64bedeeb102f0e0f15d793da9d3f4ed718fbbef]

Error: authorization.RoleAssignmentsClient#Create: Failure responding to request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=409 Code="RoleAssignmentExists" Message="The role assignment already exists."

  on .terraform/modules/keyvault_flexvolume_role/cluster/azure/keyvault_flexvol_role/main.tf line 5, in resource "azurerm_role_assignment" "flexvol":
   5: resource "azurerm_role_assignment" "flexvol" {
```

**To Reproduce:**

- Deploy `azure-common-infra` environment
- Deploy `azure-single-keyvault` environment
- Try to deploy `azure-single-keyvault-cosmos-mongo-db-simple` but throwing above error.

**Expected behavior:**

Be able to use single key vault in multiple environments.

**Does this require updates to documentation?:**

Maybe

**Comment by andrebriggs** [comment]
@sarath-p was this only happening with azure-single-keyvault-cosmos-mongo-db-simple ?

**Comment by sarath-p** [comment]
@andrebriggs - that's correct but we concluded that it is expected. During testing, I was using the same `Service Principal` and common `Key Vault` for all templates/environments. In this scenario, the `Role Assignment` is created with the first template deployment and subsequent deployments gets this error. 

We can close this issue with an update in the `docs` that `Service Principal` can't be shared between deployments in at a tenant level.

**Comment by jmspring** [comment]
closed by PR #1419
